src/main/java/org/variantsync/diffdetective/diff/difftree/DiffType.java

```diff
@@ -2,6 +2,8 @@
 
 import java.util.function.Consumer;
 
+import org.apache.commons.lang3.function.FailableConsumer;
```

Comment:
We also have interfaces for this in `Functjonal`. They are called `FragileSupplier`... I did not know that Apache Commons also had these. It would probably be good to be consistent and always use the same interfaces? The Functjonal-Interfaces also have support within Functjonal, for example for the `Result` type. What is your opinion?

Reply:
I checked their source code: These interfaces are almost identical. The only difference is that Functjonal uses `Exception` as bound and Apache Commons Lang3 uses `Throwable`, which is more general.

I propose to
- remove the `functions` package from Functjonal
- add a Apache Commons Lang3 dependency to Functjonal
- use `Failable*` instead of `Fragile*` everywhere

Pro:
- Java has a nominal type system, so two identical interfaces are not compatible. That will probably result in pain as soon as some dependency of DiffDetective uses Apache Commons Lang3.
- Apache Commons Lang3 already has failable alternatives for most (probably all, but I didn't check that) of the standard functional interfaces in `java.util.function`.
- Apache Commons Lang3 also provides some useful helpers, for example converting a `FailableX` to an `X` by rethrowing all exceptions as `RuntimeException`. It also provides a `FailableStream`.

Cons:
- Adding a Apache Commons Lang3 dependency to Functjonal.
- This is a breaking change for users of Functjonal. I don't know how many users Functjonal has, so I can't estimating how much impact that would have.